### ISSUE TITLE
java.lang.String cannot be cast to java.lang.Boolean when uploading telemetry data using the HTTP adapter

### ISSUE THREAD

**Issue by maximilianschwab** [eclipse/hono#547]
I setup Hono on Docker Swarm using the tutorial on this site:
https://www.eclipse.org/hono/getting-started/

Registering a device works perfect but every time when i try to upload telemetry for the example device with the id 4711 i get the following response:

HTTP/1.1 503 Service Unavailable
Content-Length: 19
Retry-After: 2

Service Unavailable

When i try to update the telemetry i see the following log of the REST adapter.

```hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:44:08.505 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.RequestResponseClientConfigProperties - using certificates from file [/run/secrets/trusted-certs.pem] as trust anchor
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:44:08.505 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.ConnectionFactoryImpl - connecting to AMQP 1.0 container [amqps://hono-service-device-registry.hono:5671]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:44:08.774 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.ConnectionFactoryImpl - connected to AMQP 1.0 container [amqps://hono-service-device-registry.hono:5671], opening connection ...
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:44:08.779 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.ConnectionFactoryImpl - connection to container [Hono-DeviceRegistry-0.0.0.0:5671] at [amqps://hono-service-device-registry.hono:5671] open
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:44:08.779 [vert.x-eventloop-thread-1] INFO  o.e.h.s.a.d.UsernamePasswordAuthProvider - reconnected to Credentials service
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.283 [vert.x-eventloop-thread-1] DEBUG o.e.hono.client.impl.HonoClientImpl - creating new client for credentials/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.285 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.impl.CredentialsClientImpl - creating new credentials client for [DEFAULT_TENANT]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.307 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.AbstractHonoClient - receiver open [source: credentials/DEFAULT_TENANT/5a6af998-d0c9-412e-9fa8-ff9c2340be39]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.310 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.AbstractHonoClient - sender open [target: credentials/DEFAULT_TENANT, sendQueueFull: false]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.311 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - request-response client for peer [Hono-DeviceRegistry-0.0.0.0:5671] created
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.311 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.impl.CredentialsClientImpl - successfully created credentials client for [DEFAULT_TENANT]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.311 [vert.x-eventloop-thread-1] DEBUG o.e.hono.client.impl.HonoClientImpl - successfully created new client for credentials/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.317 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: credentials/DEFAULT_TENANT, subject: get, correlation ID: cred-client-ad83a9b2-8c6f-4e9a-a7db-c6b7d2c21f38] to service
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.330 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: credentials/DEFAULT_TENANT/5a6af998-d0c9-412e-9fa8-ff9c2340be39, subject: null, correlation ID: cred-client-ad83a9b2-8c6f-4e9a-a7db-c6b7d2c21f38, status: 200]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.335 [vert.x-eventloop-thread-2] DEBUG o.e.hono.client.impl.HonoClientImpl - creating new message sender for telemetry/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.335 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - creating new client for registration/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.336 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.impl.RegistrationClientImpl - creating new registration client for [DEFAULT_TENANT]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.336 [vert.x-eventloop-thread-0] INFO  o.e.h.c.i.AbstractRequestResponseClient - enabling caching of responses from registration/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.344 [vert.x-eventloop-thread-0] DEBUG o.e.h.client.impl.AbstractHonoClient - receiver open [source: registration/DEFAULT_TENANT/2ef702e8-5791-4459-ba48-0864b6e630d7]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.347 [vert.x-eventloop-thread-0] DEBUG o.e.h.client.impl.AbstractHonoClient - sender open [target: registration/DEFAULT_TENANT, sendQueueFull: false]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.348 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - request-response client for peer [Hono-DeviceRegistry-0.0.0.0:5671] created
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.348 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.impl.RegistrationClientImpl - successfully created registration client for [DEFAULT_TENANT]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.348 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - successfully created new client for registration/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.365 [vert.x-eventloop-thread-1] DEBUG o.e.hono.client.impl.HonoClientImpl - creating new client for tenant
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.365 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.TenantClientImpl - creating new tenant API client.
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.366 [vert.x-eventloop-thread-1] INFO  o.e.h.c.i.AbstractRequestResponseClient - enabling caching of responses from tenant/null
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.367 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: registration/DEFAULT_TENANT, subject: assert, correlation ID: reg-client-1fb8f407-d6f0-4dc8-9bbd-0bb28474035c, device ID: 4711] to service
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.368 [vert.x-eventloop-thread-2] DEBUG o.e.h.client.impl.AbstractHonoClient - sender open [target: telemetry/DEFAULT_TENANT, sendQueueFull: true]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.370 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: registration/DEFAULT_TENANT/2ef702e8-5791-4459-ba48-0864b6e630d7, subject: null, correlation ID: reg-client-1fb8f407-d6f0-4dc8-9bbd-0bb28474035c, status: 200]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.378 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.AbstractHonoClient - receiver open [source: tenant/null/c60a2f60-cf6c-4ecd-964a-e1e9c83469ab]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.381 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.AbstractHonoClient - sender open [target: tenant/null, sendQueueFull: false]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.381 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - request-response client for peer [Hono-DeviceRegistry-0.0.0.0:5671] created
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.382 [vert.x-eventloop-thread-1] DEBUG o.e.h.client.impl.TenantClientImpl - successfully created tenant API client
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.382 [vert.x-eventloop-thread-1] DEBUG o.e.hono.client.impl.HonoClientImpl - successfully created new client for tenant
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.385 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: tenant/null, subject: get, correlation ID: tenant-client-39fd640d-36ad-4ad2-8dbe-0d88a928cfd6] to service
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.386 [vert.x-eventloop-thread-1] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: tenant/null/c60a2f60-cf6c-4ecd-964a-e1e9c83469ab, subject: null, correlation ID: tenant-client-39fd640d-36ad-4ad2-8dbe-0d88a928cfd6, status: 200]
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.389 [vert.x-eventloop-thread-2] DEBUG o.e.h.client.impl.AbstractHonoClient - sender [target: telemetry/DEFAULT_TENANT] has 250 credits after grace period of 20ms
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.390 [vert.x-eventloop-thread-2] DEBUG o.e.hono.client.impl.HonoClientImpl - successfully created new message sender for telemetry/DEFAULT_TENANT
hono-adapter-http-vertx.1.ro81e31oiz1g@v22016103605938256    | 15:49:14.390 [vert.x-eventloop-thread-2] DEBUG o.e.h.a.h.AbstractVertxBasedHttpProtocolAdapter - cannot process message for device [tenantId: DEFAULT_TENANT, deviceId: 4711, endpoint: telemetry]: java.lang.String cannot be cast to java.lang.Boolean
```

The last line shows an casting error -> java.lang.String cannot be cast to java.lang.Boolean

Has someone a hint to solve the problem? Any helping answer to solve the problem is appreciated.

**Comment by sysexcontrol** [comment]
Could reproduce it just now, let me have a look...

**Comment by sysexcontrol** [comment]
Fixed with 41e7d31.
Thanks for reporting!
